crypto/libsodium-fork/src/libsodium/crypto_sign/ed25519/ref10/open.c

```diff
@@ -10,6 +10,30 @@
 #include "private/ed25519_ref10.h"
 #include "utils.h"
 
+
+int 
+validate_ed25519_pk_and_sig(const unsigned char *sig, const unsigned char *pk)
+{
+#ifdef ED25519_COMPAT
+    if (sig[63] & 224) {
+        return -1;
+    }
+#else
+    if (sc25519_is_canonical_vartime(sig + 32) == 0)  {
+        return -1;
+    }
+    if (ge25519_is_canonical_vartime(sig) == 0 ) {
+        return -1;
+    }
+
+    if (ge25519_is_canonical_vartime(pk) == 0 ||
+        ge25519_has_small_order(pk) != 0) {
+        return -1;
+    }
+#endif
+    return 0;
+}
+
 int
 _crypto_sign_ed25519_verify_detached(const unsigned char *sig,
```

Comment:
Consider splitting out into another exported function.

Reply:
how would you suggest I should split it?
could it be reused?
I guess I'm missing somthing.